lib/engine.js

```diff
@@ -393,77 +397,59 @@ class Engine {
-
-        const downloadAsset = !disableInputDownload && !generateUrl;
-
         // create input object (type Asset)
         let transformerInput = {};
-        // asset needs to be downloaded locally
-        if(downloadAsset && input.url && !input.path) {
-
-            this[INTERNAL].timers.download.start();
-
-            // download input to <baseDirectory>/<stepNumber>-<transformerName>/inputfile and set input.path
-            transformerInput = await getSource(
-                input.url,
-                transformerDirectory.base,
-                false
-            );
-
-            this[INTERNAL].timers.download.stop();
-
-        } // url to asset is available and transformer to use it to process asset (ex: worker-ffmpeg)
-        else if(!downloadAsset && input.url) {
-            // dont download - T.compute will use url to process asset
-            // input.url is already set and is passed on to Asset object 
-            debug(`asset url available and disableInputDownload is ${disableInputDownload} : dont need to do anything`);
-            transformerInput = new Asset(input);
-        } // if (file exists locally && (T.manifest.input.sourceType === 'URL') then upload to tmp cloud
-        else if(!downloadAsset && input.path) {
-            const preSignedUrl = await this.getPresignedUrl(input.path);
-            // TODO: remove this log
-            debug('generated presigned url', preSignedUrl);
-            input.url = preSignedUrl;
-            transformerInput = new Asset(input, input.path);
-
-        } // asset is an intermediate rendition from previous transformer and is already available in input.path
-        else if(input.path) {
-            transformerInput = new Asset(input, input.path);
+        if (input && !input.url && !input.path) {
+            throw new GenericError('No source file accessible.', 'prepareTransformer');
         }
 
-        if (!transformerInput || !transformerInput.path) {
-            throw new GenericError('No source file accessible.');
+        const inputUrl = input.url || '';
+        let protocol = '';
+        if (inputUrl) {
+            if (!validDataUrl(inputUrl) && !validUrl.isHttpsUri(inputUrl)) {
+                throw new SourceUnsupportedError(`${inputUrl} must be a valid https url or datauri`);
+            }
+            protocol = new URL(inputUrl).protocol;
+        }
+    
+        debug(`sourceType: ${input.sourceType}`);
+        if (input.sourceType === TransformerSourceType.URL) {
+            if (inputUrl && (protocol === 'https:')) {
+                debug(`asset url available : dont need to do anything`);
+                transformerInput = new Asset(input);
+            } else if (inputUrl && protocol === 'data:') {
+                debug(`asset url available as data URI : upload asset to a temp cloud storage and generate a presigned URL`);
+                transformerInput = await getSource(input, transformerDirectory.base, true);
```

Comment:
why are we using getSource instead of getAsset?

Reply:
it was the original func being used, and I think getSource also runs `const name = sourceFilename(paramsSource);` which is a param of getAsset.